File: mario.py
from cs50 import get_int


def main():
    # Using a while loop to get a positive integer
    # Loop will break only when input will be an int between 0 and 8
    while True:
        height = get_int("Give me a number bro, I'll make a pyramid! Input height: ")
        width = height
        if height >= 0 and height <= 8:
            break

    # In range function, 1 is the start point and height + 1 is the end point
    for i in range(1, height + 1):

        # Hashes to print are equal to the row number
        hashes = i
        # In each row the sum of hashes and spaces is equal to the user's input
        spaces = width - hashes

        # Printing spaces without new line
        print(" " * spaces, end="")
        # Printing hashes without new line
        print("#" * hashes, end="")

        # Printing two spaces
        print("  ", end="")

        # Printing the other half of the pyramid which doesn't require spaces
        print("#" * hashes)


if __name__ == "__main__":
    main()

File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    //declaring my height variable which I want to retrieve from the user with the get_int functio
    int height;

    //using a do-while loop so that I can prompt the user again and again the same question until I get an integer that meets my condition
    do
        {
        height = get_int("How tall should be the pyramid? Give me a number from 1 to 8.\n");
         }
    //this is my condition
    while (height > 8 || height < 1);

    //declaring three variable because the pyramid is built with spaces, hashtags and going to next row
    int i;
    int j;
    int k;

    //Using three nested for loops
    //This loop prints out a new line. The numbers of lines printed out must be equal to the height of the pyramid
    //Hence my condition is linked to the height input 
    for (i = 0; i < height; i++)
    {
    
    //This loop prints the spaces. The number of spaces I need to print per row depends on the height of the pyramid
    //Hence the variable that initializes the loop is set to be equal to the height - 1
    for (k = height - 1 ; k > i ; k--)
        {
            printf(" ");
        }
    
    //This loop is printing #. The number of # I have to print per row is equal to the row number
    //So I'm tying this loop condition to the variable that initializes the for loop printing out new lines
    for (j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}
